frontend/src/app/[locale]/subscribe/actions.ts

```diff
@@ -0,0 +1,134 @@
+"use server";
+
+import { environment } from "src/constants/environments";
+import { TFn } from "src/types/intl";
+import { z } from "zod";
+
+import { getTranslations } from "next-intl/server";
+import { redirect } from "next/navigation";
+
+import { ValidationErrors } from "src/components/subscribe/SubscriptionForm";
+
+type sendyResponse = {
+  validationErrors: ValidationErrors;
+  errorMessage: string;
+};
+
+export async function subscribeEmail(
+  prevState: sendyResponse,
+  formData: FormData,
+) {
+  const t = await getTranslations();
+
+  const { errorMessage, validationErrors } = await subscribeEmailAction(
+    t,
+    formData,
+  );
+  if (errorMessage ?? validationErrors) {
+    return {
+      errorMessage,
+      validationErrors,
+    };
+  }
+  // Navigate to the sub confirmation page if no error returns short circuit the function
+  redirect(`/subscribe/confirmation`);
+}
+
+export async function subscribeEmailAction(
+  t: TFn,
```

Comment:
I wonder if the real solution here is to just pass along the strings that the function needs rather than passing down the entire "t" function. That would likely mean passing a lot of strings, though, or basically partially recreating the messaging object in order to pass through just what we need. In any case I'm still worried about the entire message object getting sent down to the client, but not enough to continue blocking this PR, especially since I don't see an immediately ideal solution otherwise.